File: Communicator/Communicator.DataProvider/Models/Channel.cs
using Newtonsoft.Json;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;


namespace Communicator.DataProvider.Models
{
    public class Channel
    {
        public int ChannelId { get; set; }
        public string ChannelName { get; set; }
        public bool isGroupChannel { get; set; }
        public ICollection<Message> Messages { get; set; }
        [JsonIgnore]
        public ICollection<ApplicationUserChannel> ApplicationUserChannels { get; set; }

        [NotMapped]
        public string[] UserIds
        {
            get { return ApplicationUserChannels.Select(x => x.UserId).ToArray(); }
        }

    }
}

File: Communicator/Communicator.Website/ClientApp/src/store/Models/Channel.ts
import { Message } from "./Message";
export class Channel {
    constructor(channelId:number) {
        this.ChannelId = channelId;
    }
    UserIds!: string[];
    ChannelName!: string;
    ChannelId: number;
    Messages!: Message[];
    isGroupChannel!: boolean;
}

